### PR TITLE
Create a single universal executable for macOS

### LINKED ISSUE #946: Add Mac arm64 release
### Describe the request

We should start to build and distribute arm64 build for macOS.

- [x] Update the build process and CI to output a [Universal macOS binary](https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary) (that includes both the Intel and arm64 executables)
- [ ] Ensure the auto-update process works correctly on both architectures
- [x] Update the README.md file section about Apple Silicon support

### Describe the current behavior

Arm64 compatibility is offered through Rosetta emulation. See [Readme.md](https://github.com/arduino/arduino-create-agent?tab=readme-ov-file#apple-silicon-support)

### Issue checklist

- [X] I searched for previous requests in [the issue tracker](https://github.com/arduino/arduino-create-agent/issues?q=)
- [X] I verified the feature was still missing when using the latest version
- [X] My request contains all necessary details

### PR DESCRIPTION
**Pull request by MatteoPologruto**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-create-agent/pulls)
      before creating one)
- [ ] Tests for the changes have been added (for bug fixes / features)

* **What kind of change does this PR introduce?**

Instead of relying on the Rosetta translation to distribute the Agent for arm architectures, a single universal executable is created. It runs natively on both Apple silicon and Intel-based Mac computers.

- **Does this PR introduce a breaking change?**

No

* **Other information**:

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -107,21 +107,40 @@ jobs:
...
+          CGO_CFLAGS: -mmacosx-version-min=10.15
+          CGO_LDFLAGS: -mmacosx-version-min=10.15
+          GOARCH: arm64
+        run: |
+          task go:build
+          mv ${{ env.PROJECT_NAME }} ${{ env.PROJECT_NAME }}_arm64
+        if: matrix.os == 'macos-12'
+
+      - name: Create universal macos executable
+        shell: zsh
```
Comment (umbynos) [inline .github/workflows/release.yml:134]: is this shell required? :upside_down_face:

Comment (MatteoPologruto) [reply .github/workflows/release.yml]: nope

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -107,21 +107,41 @@ jobs:
...
+          rm ${{ env.PROJECT_NAME }}_amd64 ${{ env.PROJECT_NAME }}_arm64
         if: matrix.os == 'macos-12'
 
         # this will create `public/` dir with compressed full bin (<version>/<os>-<arch>.gz) and a json file
       - name: Create autoupdate files
         run: go-selfupdate ${{ env.PROJECT_NAME }}${{ matrix.ext }} ${TAG_VERSION}
         if: matrix.arch != '386' && steps.prerelease.outputs.IS_PRE != 'true'
 
-        # for now we do not distribute m1 build, this is a workaround for now
+      # for now we do not distribute m1 build, this is a workaround for now
```
Comment (umbynos) [inline .github/workflows/release.yml:144]:
```suggestion
```
I think this can be removed now

**Comment by umbynos** [comment]
Have you had the chance to test the autoupdate mechanism?
Another doubt that comes to my mind is: have you tested that when using the cloud editor the tools install is performed successfully? (selecting the correct macos arm64 tool if present)?

**Comment by Xayton** [comment]
The tests were good:
- The auto-update mechanism works correctly. The tests found no issues (but an opportunity for improvement).
- The tools installation was successful: the arm64 version of the executable will try to download the "closes" version, so arm64 if available, or amd64 otherwise. In both cases the download works.

**Review by Xayton — APPROVED** [review]

**Pull request merged**